File: Data Structures & Algorithms/validate-parentheses/submission-1.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<Character>();
        for(Character c : s.toCharArray()){
            if (c.equals('(') || c == '['|| c.equals('{')){
                stack.push(c);
            }
            else{
                if(stack.isEmpty()) return false;
                Character popped = stack.pop();
                if(popped.equals('(') && !c.equals(')')){
                    return false;
                }
                else if(popped.equals('{') && !c.equals('}')){
                    return false;
                }
                else if(popped.equals('[') && !c.equals(']')){
                    return false;
                }
            }
        }
        return stack.isEmpty();
    }
}

File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        let = {}
        if len(s) != len(t):
            return False
        for c in s:
            if let.get(c):
                let[c] = let[c] + 1
            else:
                let[c] = 1
        for c in t:
            if let.get(c) and let[c] > 0:
                let[c] = let[c] - 1
        for out in let.values():
            if out != 0:
                return False
        return True

        

        